src/Quaternion.jl

```diff
@@ -293,3 +293,25 @@ function slerp(qa::Quaternion{T}, qb::Quaternion{T}, t::T) where {T}
         qa.v3 * ratio_a + qm.v3 * ratio_b,
     )
 end
+
+function _complex_representation(A::Matrix{Quaternion{T}}) where {T}
+	# convert a quatenion matrix to corresponding complex matrix
+	a = map(t->t.s, A)
+    b = map(t->t.v1, A)
+    c = map(t->t.v2, A)
+    d = map(t->t.v3, A)
+    
+	return [a+im*b c+im*d;-c+im*d a-im*b]
+end
+
+function _quaternion_representation(A::Matrix{Complex{T}}) where {T}
+	n = round(Int, size(A, 1)/2)
+    a = real(A[1:n, 1:n])
```

Comment:
Might be better to use a `view` (`@view A[1:n, 1:n]`) and then call `real` / `imag` on the scalar in the `Quaternion` call below.